Combat/Source/Points.cpp

```diff
@@ -0,0 +1,111 @@
+#include "Points.hpp"
+#include <cstdlib>
+
+Points::Points(const uint maxPoints)
+{
+   _maxPoints = maxPoints;
+   _currentPoints = maxPoints;
+}
+
+Points::Points(const uint maxPoints, const uint currentPoints)
+{
+      _maxPoints = maxPoints;
+
+      if(currentPoints > maxPoints)
+      {
+         _currentPoints = maxPoints;
+      }
+      else
+      {
+         _currentPoints = currentPoints;
+      }
+}
+
+// Sprawdzanie na karcie czy currentMP >= amount
+void Points::remove(const uint amount)
+{
+   if(amount > _currentPoints)
+   {
+      _currentPoints = 0;
+   }
+   else
+   {
+      _currentPoints -= amount;
+   }
+}
+
+void Points::add(const uint amount)
+{
+   if(_currentPoints + amount > _maxPoints)
```

Comment:
Jak zrobić to szybciej:
```
_currentPoints += amount;
if(_currentPoints > _maxPoints)
{
   _currentPoint = _maxPoints;
}
```

Teraz przejdźcie każdą możliwą ścieżką i zobaczcie ile operacji za każdym razem robicie w oobu przypadkach ;)

(to są szczegóły, ale jestem spaczony przez to, że duzo robiłem w embedded, Wasz kod też jest spoko i czytelny!)